extensions/amp-analytics/0.1/test/test-amp-analytics.js

```diff
@@ -250,13 +250,42 @@ describes.realWin('amp-analytics', {
     }
   });
 
-  it('preload script', function() {
+  it('does not unnecessarily preload iframe transport script', function() {
     const el = doc.createElement('amp-analytics');
     doc.body.appendChild(el);
     const analytics = new AmpAnalytics(el);
+    sandbox.stub(analytics, 'assertAmpAdResourceId').callsFake(() => 'fakeId');
     analytics.buildCallback();
     analytics.preconnectCallback();
-    return Promise.resolve().then(() => {
+    return analytics.layoutCallback().then(() => {
+      const preloads = doc.querySelectorAll('link[rel=preload]');
+      expect(preloads).to.have.length(0);
+    });
+  });
+
+  it('preloads iframe transport script if relevant', function() {
+    const el = doc.createElement('amp-analytics');
+    el.setAttribute('type', 'foo');
+    doc.body.appendChild(el);
+    const analytics = new AmpAnalytics(el);
+    sandbox.stub(analytics, 'assertAmpAdResourceId').callsFake(() => 'fakeId');
+    analytics.predefinedConfig_['foo'] = {
+      'transport': {
+        'iframe': 'https://www.google.com',
+      },
+      'triggers': {
+        'sample_visibility_trigger': {
+          'on': 'visible',
+          'request': 'sample_visibility_request',
+        },
+      },
+      'requests': {
+        'sample_visibility_request': 'fake_request',
+      },
+    };
+    analytics.buildCallback();
+    analytics.preconnectCallback();
+    return analytics.layoutCallback().then(() => {
```

Comment:
Should have asked this in the review for #12050 but does anyone know if the 3 lines below this comment are expected to succeed in Safari <=10?

Reply:
Good question! I wouldn't expect them to.  I recommend you mock out call to this.preconnect.preload that way your check isn't dependent on implementation details and will ensure coverage for Safari